File: ui/visual-component/src/main/java/org/visual/component/display/container/RoutePane.java
package org.visual.component.display.container;

import java.util.concurrent.ConcurrentHashMap;
import javafx.scene.layout.Pane;
import javafx.scene.layout.StackPane;

public class RoutePane extends StackPane {
  private final ConcurrentHashMap<String, Integer> routers = new ConcurrentHashMap<>();
  private int index = 0;

  public void addRoute(String path, Pane pane) {
    int currentIndex = getChildren().isEmpty() ? 0 : -1;
    getChildren().add(currentIndex, pane);
    routers.put(path, currentIndex);
  }

  public void next() {
    // Add your implementation for the 'next' method if needed
  }
}

File: ui/visual-model-component/src/main/kotlin/org/visual/model/component/container/RoutePane.kt
package org.visual.model.component.container

import java.util.concurrent.ConcurrentHashMap
import javafx.scene.layout.Pane
import javafx.scene.layout.StackPane
import org.visual.model.component.annotation.FxComponent

@FxComponent
class RoutePane : StackPane() {
  private val routers by lazy { ConcurrentHashMap<String, Int>() }

  fun addRoute(path: String, pane: Pane) {
    val index = if (children.isEmpty()) 0 else -1
    children.add(index, pane)
    routers[path] = index
  }
}
